tools/analyze_cms_sizes.py

```diff
@@ -0,0 +1,281 @@
+    
+    for path in new_files:
+        if path in old_files:
+            new_size = new_files[path]
+            old_size = old_files[path]
+            
+            if old_size > 0:
+                percentage_change = ((new_size - old_size) / old_size) * 100
+                changes.append({
+                    'path': path,
+                    'new_size': new_size,
+                    'old_size': old_size,
+                    'absolute_change': new_size - old_size,
+                    'percentage_change': percentage_change
+                })
+    
+    return changes
+
+def find_outliers(changes, top_n=10):
+    """Find the top N files with largest percentage increases and decreases."""
+    increases = sorted(changes, key=lambda x: x['percentage_change'], reverse=True)[:top_n]
+    decreases = sorted(changes, key=lambda x: x['percentage_change'])[:top_n]
+    
+    return increases, decreases
+
+def create_histogram(percentage_changes, bins=10, title="Size Change Distribution", show_stats=True, show_detailed_for_first_bucket=False):
+    """Create a text-based histogram of percentage changes."""
+    if not percentage_changes:
+        return
+    
+    min_change = min(percentage_changes)
+    max_change = max(percentage_changes)
+    
+    # Create bins
+    bin_width = (max_change - min_change) / bins
+    bin_counts = [0] * bins
+    bin_ranges = []
+    
+    for i in range(bins):
+        bin_start = min_change + i * bin_width
+        bin_end = min_change + (i + 1) * bin_width
+        bin_ranges.append((bin_start, bin_end))
+    
+    # Count occurrences in each bin
+    for change in percentage_changes:
+        if change >= max_change:  # Handle edge case for max value
+            bin_index = bins - 1
+        else:
+            bin_index = int((change - min_change) / bin_width)
+        bin_counts[bin_index] += 1
+    
+    # Find max count for scaling
+    max_count = max(bin_counts) if bin_counts else 0
+    if max_count == 0:
+        return
+    
+    # Create histogram
+    if "Detailed" in title:
+        print(f"\n{title} ({len(percentage_changes)} files):")
+    else:
+        print(f"\n{title}:")
+    print(f"Range: {min_change:.1f}% to {max_change:.1f}%")
+    
+    # Scale to 40 characters max width
+    scale = 40 / max_count
+    
+    for i, (count, (bin_start, bin_end)) in enumerate(zip(bin_counts, bin_ranges)):
+        bar_length = int(count * scale)
+        bar = '█' * bar_length
+        percentage = (count / len(percentage_changes)) * 100
+        print(f"{bin_start:6.1f}% - {bin_end:6.1f}%: {count:3d} ({percentage:4.1f}%) {bar}")
+    
+    # Statistical summary
+    if show_stats:
+        mean_change = sum(percentage_changes) / len(percentage_changes)
+        sorted_changes = sorted(percentage_changes)
+        median_change = sorted_changes[len(sorted_changes) // 2]
+        print(f"\nMedian: {median_change:.2f}%, Std dev: {calculate_std_dev(percentage_changes, mean_change):.1f}%")
```

Comment:
I'd move `calculate_std_dev` to before `create_histogram` since it's used in it.